src/main/java/com/google/sps/servlets/course/StudentParticipants.java

```diff
@@ -0,0 +1,70 @@
+package com.google.sps.servlets.course;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.gson.Gson;
+import com.google.sps.firebase.FirebaseAppManager;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/student-participants")
+public class StudentParticipants extends HttpServlet {
```

Comment:
This servlet can be combined with `TaParticipants`. The combined servlet could take a parameter `role` that would be student, ta, or owner. Then the servlet could return the correct users.

This will reduce code duplication because of how similar `StudentParticipants` and `TaParticipants` are.

Reply:
Combined